var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -187,3 +188,168 @@ def cmake_args(self):
+#elif defined(VTKM_ENABLE_TBB)
+    constexpr vtkm::cont::DeviceAdapterTagTBB desired_device;
+#elif defined(VTKM_ENABLE_OPENMP)
+    constexpr vtkm::cont::DeviceAdapterTagOpenMP desired_device;
+#else
+    #error "No VTK-m Device Adapter enabled"
+#endif
+
+    std::cout << "-------------------------------------------\n";
+    std::cout << "Testing No Argument Kernel" << std::endl;
+    vtkm::cont::Algorithm::Schedule(desired_device, NoArgKernel(), size);
+
+    vtkm::cont::ArrayHandle<vtkm::Id> handle;
+    {
+    std::cout << "-------------------------------------------\n";
+    std::cout << "Testing Kernel + ArrayHandle PrepareForOutput" << std::endl;
+    vtkm::cont::Token token;
+    auto portal = handle.PrepareForOutput(size, desired_device, token);
+    vtkm::cont::Algorithm::Schedule(desired_device,
+        FillArrayKernel<decltype(portal)>{portal, 1}, size);
+    }
+
+    {
+    std::cout << "-------------------------------------------\n";
+    std::cout << "Testing Kernel + ArrayHandle PrepareForInPlace" << std::endl;
+    vtkm::cont::Token token;
+    auto portal = handle.PrepareForInPlace(desired_device, token);
+    vtkm::cont::Algorithm::Schedule(desired_device,
+        FillArrayKernel<decltype(portal)>{portal, 2}, size);
+    }
+
+    std::cout << "-------------------------------------------\n";
+    std::cout << "Ran tests on: " << desired_device.GetName() << std::endl;
+
+    return 0;
+}
+"""
+
+            cmakelists = r"""
+##============================================================================
+##  Copyright (c) Kitware, Inc.
+##  All rights reserved.
+##  See LICENSE.txt for details.
+##
+##  This software is distributed WITHOUT ANY WARRANTY; without even
+##  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+##  PURPOSE.  See the above copyright notice for more information.
+##============================================================================
+cmake_minimum_required(VERSION 3.12...3.18 FATAL_ERROR)
+project(VTKmSmokeTest CXX)
+
+#Find the VTK-m package
+find_package(VTKm REQUIRED QUIET)
+
+add_executable(VTKmSmokeTest main.cxx)
+target_link_libraries(VTKmSmokeTest PRIVATE vtkm_cont)
+vtkm_add_target_information(VTKmSmokeTest
+                            DROP_UNUSED_SYMBOLS MODIFY_CUDA_FLAGS
+                            DEVICE_SOURCES main.cxx)
+"""
+
+            with open("main.cxx", 'w') as f:
+                f.write(source)
+            with open("CMakeLists.txt", 'w') as f:
+                f.write(cmakelists)
+            builddir = "build"
+            with working_dir(builddir, create=True):
+                cmake = Executable(spec['cmake'].prefix.bin + "/cmake")
+                cmakefiledir = spec['vtk-m'].prefix.lib + "/cmake"
+                cmakefiledir = cmakefiledir + "/" + os.listdir(cmakefiledir)[0]
+                cmake(*(["..", "-DVTKm_DIR=" + cmakefiledir]))
+                cmakebuild = Executable('cmake --build .')
```

Comment:
do we need to pass this as a list to `Executable`, like what is done one line above?

Reply:
That's a good point. I have changed it to: `cmake(*(["--build", "."]))`